### PR TITLE
Add label, date and datetime picker to system.xml

### PR DESCRIPTION
**Pull request by luigifab**
### Description

Some new fields for system.xml, example:

image

XML of the screen shot:
```xml
<head>
	<label>Crash test</label>
	<frontend_model>adminhtml/system_config_form_field_heading</frontend_model>
	<sort_order>1001</sort_order>
	<show_in_default>1</show_in_default>
	<show_in_website>1</show_in_website>
	<show_in_store>1</show_in_store>
</head>
<label_1>
	<label>label</label>
	<value>Pouet</value>
	<frontend_model>adminhtml/system_config_form_field_label</frontend_model>
	<sort_order>1002</sort_order>
	<show_in_default>1</show_in_default>
	<show_in_website>1</show_in_website>
	<show_in_store>1</show_in_store>
</label_1>
<labelbold_2>
	<label>label bold</label>
	<value>Pouet Pouet</value>
	<bold>1</bold>
	<frontend_type>label</frontend_type><!-- optionnal and useless -->
	<frontend_model>adminhtml/system_config_form_field_label</frontend_model>
	<sort_order>1003</sort_order>
	<show_in_default>1</show_in_default>
	<show_in_website>1</show_in_website>
	<show_in_store>1</show_in_store>
</labelbold_2>
<datetime_1>
	<label>datetime (original)</label>
	<frontend_model>adminhtml/system_config_form_field_datetime</frontend_model>
	<sort_order>1004</sort_order>
	<show_in_default>1</show_in_default>
	<show_in_website>1</show_in_website>
	<show_in_store>1</show_in_store>
</datetime_1>
<datetime_2>
	<label>datetime with format</label>
	<frontend_type>datetime</frontend_type><!-- optionnal and useless -->
	<frontend_model>adminhtml/system_config_form_field_datetime</frontend_model>
	<format>FORMAT_TYPE_SHORT</format>
	<sort_order>1005</sort_order>
	<show_in_default>1</show_in_default>
	<show_in_website>1</show_in_website>
	<show_in_store>1</show_in_store>
</datetime_2>
<date_3>
	<label>date</label>
	<frontend_model>adminhtml/system_config_form_field_date</frontend_model>
	<sort_order>1006</sort_order>
	<show_in_default>1</show_in_default>
	<show_in_website>1</show_in_website>
	<show_in_store>1</show_in_store>
</date_3>
<date_4>
	<label>date with format</label>
	<frontend_type>date</frontend_type><!-- optionnal and useless -->
	<frontend_model>adminhtml/system_config_form_field_date</frontend_model>
	<format>FORMAT_TYPE_SHORT</format>
	<sort_order>1007</sort_order>
	<show_in_default>1</show_in_default>
	<show_in_website>1</show_in_website>
	<show_in_store>1</show_in_store>
</date_4>
<datetime_5>
	<label>editable datetime</label>
	<frontend_model>adminhtml/system_config_form_field_datetime</frontend_model>
	<backend_model>adminhtml/system_config_backend_datetime</backend_model>
	<editable>1</editable>
	<sort_order>1008</sort_order>
	<show_in_default>1</show_in_default>
	<show_in_website>1</show_in_website>
	<show_in_store>1</show_in_store>
</datetime_5>
<datetime_6>
	<label>editable datetime with format</label>
	<frontend_type>datetime</frontend_type><!-- optionnal and useless -->
	<frontend_model>adminhtml/system_config_form_field_datetime</frontend_model>
	<backend_model>adminhtml/system_config_backend_datetime</backend_model>
	<format>FORMAT_TYPE_SHORT</format>
	<editable>1</editable>
	<sort_order>1009</sort_order>
	<show_in_default>1</show_in_default>
	<show_in_website>1</show_in_website>
	<show_in_store>1</show_in_store>
</datetime_6>
<date_7>
	<label>editable date</label>
	<frontend_model>adminhtml/system_config_form_field_date</frontend_model>
	<backend_model>adminhtml/system_config_backend_date</backend_model>
	<editable>1</editable>
	<sort_order>1010</sort_order>
	<show_in_default>1</show_in_default>
	<show_in_website>1</show_in_website>
	<show_in_store>1</show_in_store>
</date_7>
<date_8>
	<label>editable date with format</label>
	<frontend_type>date</frontend_type><!-- optionnal and useless -->
	<frontend_model>adminhtml/system_config_form_field_date</frontend_model>
	<backend_model>adminhtml/system_config_backend_date</backend_model>
	<format>FORMAT_TYPE_SHORT</format>
	<editable>1</editable>
	<sort_order>1011</sort_order>
	<show_in_default>1</show_in_default>
	<show_in_website>1</show_in_website>
	<show_in_store>1</show_in_store>
</date_8>
```
There are some bugs with the JS calendar, sometimes it doesn't understand the format, so it's better to use the short format. But this is also another problem.

Date and datetime are stored in UTC time in `core_config_data`. If you check with Adminer, use `SET @@session.time_zone='+00:00';` (https://dba.stackexchange.com/q/20217/242650).

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All automated tests passed successfully (all builds are green)
 - [x] Add yourself to contributors list

### CODE REVIEW

**Review by fballiano — APPROVED** [review]
love some new feature

**Comment by fballiano** [comment]
but there's a conflict now

**Comment by sreichel** [comment]
@luigifab can you please check your repository settings? I cant open a PR there.

I'd to use https://github.com/luigifab/magento-lts/compare/new-config-fields...sreichel:magento-lts:new-config-fields manually.

PR: https://github.com/luigifab/magento-lts/pull/1

**Comment by luigifab** [comment]
I updated some things, not sure if the good ones.

**Comment by sreichel** [comment]
@luigifab Seems something went from? Did you pushed to my PR?

**Comment by sreichel** [comment]
> I updated some things, not sure if the good ones.
> I rebase...

Please do not force push when your previous commit is used (like in my PR). Now there are conflicts and i cant see what you recently changed. :(

**Comment by luigifab** [comment]
Oh sorry, ok I didn't check correctly what you did, I only supposed that the problem is the rebase.

**Comment by sreichel** [comment]
I use force push to, but only when i missied something in my last commit. 

At the end its more clear to use merge instead of rebase.

However .... a hope it was okay to push to your branch. If there is smomething wrong, pls let me know.

**Comment by fballiano** [comment]
Should we write something in the README? In the "what's different" section?

**Review by fballiano — DISMISSED** [review]

**Comment by fballiano** [comment]
> Should we write something in the README? In the "what's different" section?

done

**Comment by sreichel** [comment]
```
$class = (!empty($field->frontend_type) && ($field->frontend_type != 'text')) ? 'Varien_Data_Form_Element_' . ucfirst(strtolower($field->frontend_type)) : '';
```

Causes `Fatal error:  Uncaught TypeError: strtolower(): Argument #1 ($string) must be of type string, Mage_Core_Model_Config_Element given in /var/www/html/app/code/core/Mage/Adminhtml/Block/System/Config/Form/Field/Label.php:41`

**Comment by sreichel** [comment]
When saving date (w/o format) ...

```
Fatal error:  Uncaught TypeError: number_format(): Argument #1 ($num) must be of type float, string given in /var/www/html/lib/Zend/Locale/Math.php:161
Stack trace:
#0 /var/www/html/lib/Zend/Locale/Math.php(161): number_format('DEC 14, 2022 07...', 0, '.', '')
#1 /var/www/html/lib/Zend/Locale/Format.php(305): Zend_Locale_Math::floatalize('DEC 14, 2022 07...')
#2 /var/www/html/lib/Zend/Locale/Format.php(662): Zend_Locale_Format::toNumber('Dec 14, 2022 07...', Array)
#3 /var/www/html/lib/Zend/Filter/NormalizedToLocalized.php(106): Zend_Locale_Format::toFloat('Dec 14, 2022 07...', Array)
#4 /var/www/html/app/code/core/Mage/Adminhtml/Model/System/Config/Backend/Date.php(89): Zend_Filter_NormalizedToLocalized->filter('Dec 14, 2022 07...')
#5 /var/www/html/app/code/core/Mage/Adminhtml/Model/System/Config/Backend/Date.php(59): Mage_Adminhtml_Model_System_Config_Backend_Date->filterDateTime('Dec 14, 2022 07...')
```

**Comment by sreichel** [comment]
I've change some parts. Everything works, but save datetime is still not correct ...

Took fields from @luigifab post and added to cms config section ...

There is a 12h offset for datetimes ...

```
| path | value |
| :--- | :--- |
| cms/wysiwyg/datetime\_5 | 2022-12-18 03:04:01 |
| cms/wysiwyg/datetime\_6 | 2022-12-17 15:04:00 |

```

**Comment by sreichel** [comment]
I reverted my changes and tested @luigifab initial commit. It does not work. I cant save dates (same error as in my tests yesterday) and also displayed time is wrong. Now it `Dec 15, 2022 2:56:53 PM` (6h diff to correct time)

**Comment by luigifab** [comment]
I tested again with the original commit, with PHP *8.0*, with locale *fr_FR*, with browser locale *fr_FR*, with timezone utc+1, with branch *1.9.4.x*.

For me, when I save my config page with:
- `15 janv. 2023 13:32:00`, I have `2023-01-15 12:32:00` in database
- `16/01/2023 13:32`, I have `2023-01-16 12:32:00` in database
- `17 janv. 2023`, I have `2023-01-16 23:00:00` in database
- `18/01/2023`, I have `2023-01-17 23:00:00` in database

I can save my page multiple times, there are no problems.

With: `<?php echo $this->formatDate(Mage::getStoreConfig('xyz/xyz/datetime_5'), Mage_Core_Model_Locale::FORMAT_TYPE_LONG, true) ?>`, I have the good value: `15 janvier 2023 13:32:00 CET`.

But when I change OpenMage locale to *en_US*, you are right there are some bugs.

---

With your changes, I have a problem: date(time)s are not save in UTC.
For example, when I save my config page with:
- `15 janv. 2023 13:32:00`, I have `2023-01-15 13:32:00` in database, backend display `15 janv. 2023 14:32:00`
- `16/01/2023 13:32`, I have `2023-01-16 13:32:00` in database, backend display `16/01/2023 14:32`
- `17 janv. 2023`, I have `2023-01-17 00:00:00` in database, backend display `17 janv. 2023`
- `18/01/2023`, I have `2023-01-18 00:00:00` in database, backend display `18/01/2023`

With: `<?php echo $this->formatDate(Mage::getStoreConfig('xyz/xyz/datetime_5'), Mage_Core_Model_Locale::FORMAT_TYPE_LONG, true) ?>`, I have: `15 janvier 2023 14:32:00 CET`.

**Comment by fballiano** [comment]
I've tried to fix the conflicts, @luigifab can you retest it?

**Comment by fballiano** [comment]
@luigifab can you rebase it to main please?

File: app/code/core/Mage/Adminhtml/Block/System/Config/Form/Field/AbstractDate.php [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@magento.com so we can send you a copy immediately.
+ *
+ * @category   Mage
+ * @package    Mage_Adminhtml
+ * @copyright  Copyright (c) 2006-2020 Magento, Inc. (http://www.magento.com)
+ * @copyright  Copyright (c) 2022 The OpenMage Contributors (https://www.openmage.org)
```
Comment (fballiano) [inline app/code/core/Mage/Adminhtml/Block/System/Config/Form/Field/AbstractDate.php:21]:
```suggestion
 * @copyright  Copyright (c) 2023 The OpenMage Contributors (https://www.openmage.org)
```

File: app/code/core/Mage/Adminhtml/Block/System/Config/Form/Field/Date.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@magento.com so we can send you a copy immediately.
```
Comment (fballiano) [inline app/code/core/Mage/Adminhtml/Block/System/Config/Form/Field/Date.php:16]:
```suggestion
 * OpenMage
 *
 * This source file is subject to the Open Software License (OSL 3.0)
 * that is bundled with this package in the file LICENSE.txt.
 * It is also available at https://opensource.org/license/osl-3-0-php
```

File: app/code/core/Mage/Adminhtml/Block/System/Config/Form/Field/AbstractDate.php [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@magento.com so we can send you a copy immediately.
```
Comment (fballiano) [inline app/code/core/Mage/Adminhtml/Block/System/Config/Form/Field/AbstractDate.php:16]:
```suggestion
 * OpenMage
 *
 * This source file is subject to the Open Software License (OSL 3.0)
 * that is bundled with this package in the file LICENSE.txt.
 * It is also available at https://opensource.org/license/osl-3-0-php
```

File: app/code/core/Mage/Adminhtml/Block/System/Config/Form/Field/Date.php [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+ * OpenMage
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available at https://opensource.org/license/osl-3-0-php
+ *
+ * @category   Mage
+ * @package    Mage_Adminhtml
+ * @copyright  Copyright (c) 2006-2020 Magento, Inc. (http://www.magento.com)
+ * @copyright  Copyright (c) 2022 The OpenMage Contributors (https://www.openmage.org)
```
Comment (fballiano) [inline app/code/core/Mage/Adminhtml/Block/System/Config/Form/Field/Date.php:15]:
```suggestion
 * @copyright  Copyright (c) 2023 The OpenMage Contributors (https://www.openmage.org)
```

File: app/code/core/Mage/Adminhtml/Block/System/Config/Form/Field/Label.php [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@magento.com so we can send you a copy immediately.
```
Comment (fballiano) [inline app/code/core/Mage/Adminhtml/Block/System/Config/Form/Field/Label.php:16]:
```suggestion
 * OpenMage
 *
 * This source file is subject to the Open Software License (OSL 3.0)
 * that is bundled with this package in the file LICENSE.txt.
 * It is also available at https://opensource.org/license/osl-3-0-php
```

File: app/code/core/Mage/Adminhtml/Block/System/Config/Form/Field/Label.php [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+ * OpenMage
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available at https://opensource.org/license/osl-3-0-php
+ *
+ * @category   Mage
+ * @package    Mage_Adminhtml
+ * @copyright  Copyright (c) 2006-2020 Magento, Inc. (http://www.magento.com)
+ * @copyright  Copyright (c) 2022 The OpenMage Contributors (https://www.openmage.org)
```
Comment (fballiano) [inline app/code/core/Mage/Adminhtml/Block/System/Config/Form/Field/Label.php:15]:
```suggestion
 * @copyright  Copyright (c) 2023 The OpenMage Contributors (https://www.openmage.org)
```

File: app/code/core/Mage/Adminhtml/Model/System/Config/Backend/AbstractDate.php [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@magento.com so we can send you a copy immediately.
```
Comment (fballiano) [inline app/code/core/Mage/Adminhtml/Model/System/Config/Backend/AbstractDate.php:16]:
```suggestion
 * OpenMage
 *
 * This source file is subject to the Open Software License (OSL 3.0)
 * that is bundled with this package in the file LICENSE.txt.
 * It is also available at https://opensource.org/license/osl-3-0-php
```

File: app/code/core/Mage/Adminhtml/Model/System/Config/Backend/AbstractDate.php [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+ * OpenMage
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available at https://opensource.org/license/osl-3-0-php
+ *
+ * @category   Mage
+ * @package    Mage_Adminhtml
+ * @copyright  Copyright (c) 2006-2020 Magento, Inc. (http://www.magento.com)
+ * @copyright  Copyright (c) 2022 The OpenMage Contributors (https://www.openmage.org)
```
Comment (fballiano) [inline app/code/core/Mage/Adminhtml/Model/System/Config/Backend/AbstractDate.php:15]:
```suggestion
 * @copyright  Copyright (c) 2023 The OpenMage Contributors (https://www.openmage.org)
```

File: app/code/core/Mage/Adminhtml/Model/System/Config/Backend/Date.php [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@magento.com so we can send you a copy immediately.
```
Comment (fballiano) [inline app/code/core/Mage/Adminhtml/Model/System/Config/Backend/Date.php:16]:
```suggestion
 * OpenMage
 *
 * This source file is subject to the Open Software License (OSL 3.0)
 * that is bundled with this package in the file LICENSE.txt.
 * It is also available at https://opensource.org/license/osl-3-0-php
```

File: app/code/core/Mage/Adminhtml/Model/System/Config/Backend/Date.php [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+ * OpenMage
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available at https://opensource.org/license/osl-3-0-php
+ *
+ * @category   Mage
+ * @package    Mage_Adminhtml
+ * @copyright  Copyright (c) 2006-2020 Magento, Inc. (http://www.magento.com)
+ * @copyright  Copyright (c) 2022 The OpenMage Contributors (https://www.openmage.org)
```
Comment (fballiano) [inline app/code/core/Mage/Adminhtml/Model/System/Config/Backend/Date.php:15]:
```suggestion
 * @copyright  Copyright (c) 2023 The OpenMage Contributors (https://www.openmage.org)
```

File: app/code/core/Mage/Adminhtml/Model/System/Config/Backend/Datetime.php [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@magento.com so we can send you a copy immediately.
```
Comment (fballiano) [inline app/code/core/Mage/Adminhtml/Model/System/Config/Backend/Datetime.php:16]:
```suggestion
 * OpenMage
 *
 * This source file is subject to the Open Software License (OSL 3.0)
 * that is bundled with this package in the file LICENSE.txt.
 * It is also available at https://opensource.org/license/osl-3-0-php
```

File: app/code/core/Mage/Adminhtml/Model/System/Config/Backend/Datetime.php [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+ * OpenMage
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available at https://opensource.org/license/osl-3-0-php
+ *
+ * @category   Mage
+ * @package    Mage_Adminhtml
+ * @copyright  Copyright (c) 2006-2020 Magento, Inc. (http://www.magento.com)
+ * @copyright  Copyright (c) 2022 The OpenMage Contributors (https://www.openmage.org)
```
Comment (fballiano) [inline app/code/core/Mage/Adminhtml/Model/System/Config/Backend/Datetime.php:15]:
```suggestion
 * @copyright  Copyright (c) 2023 The OpenMage Contributors (https://www.openmage.org)
```

File: app/code/core/Mage/Core/Helper/Date.php [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/osl-3.0.php
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@magento.com so we can send you a copy immediately.
```
Comment (fballiano) [inline app/code/core/Mage/Core/Helper/Date.php:16]:
```suggestion
 * OpenMage
 *
 * This source file is subject to the Open Software License (OSL 3.0)
 * that is bundled with this package in the file LICENSE.txt.
 * It is also available at https://opensource.org/license/osl-3-0-php
```

File: app/code/core/Mage/Core/Helper/Date.php [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+ * OpenMage
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available at https://opensource.org/license/osl-3-0-php
+ *
+ * @category   Mage
+ * @package    Mage_Core
+ * @copyright  Copyright (c) 2006-2020 Magento, Inc. (https://www.magento.com)
+ * @copyright  Copyright (c) 2022 The OpenMage Contributors (https://www.openmage.org)
```
Comment (fballiano) [inline app/code/core/Mage/Core/Helper/Date.php:15]:
```suggestion
 * @copyright  Copyright (c) 2023 The OpenMage Contributors (https://www.openmage.org)
```

**Comment by luigifab** [comment]
Closed because this PR can be split in multiple PR, and there are issues... and I will be out there until 2026.
Anyway, I continue us it as is and the version I'm using is [here](https://www.luigifab.fr/diff/3-pull-2739-datepicker.diff) [3-pull-2739-datepicker.diff.txt](https://github.com/OpenMage/magento-lts/files/11610474/3-pull-2739-datepicker.diff.txt).

**Pull request closed without merging**